api/src/blockchains/solana.rs

```diff
@@ -176,12 +176,18 @@ impl
         let payer = Keypair::from_bytes(&self.payer_keypair)?;
         let owner = owner_address.parse()?;
 
-        let solana_collection_model = solana_collections::Entity::find()
+        let (sc, collection_model) = solana_collections::Entity::find()
+            .find_also_related(Collections)
             .filter(solana_collections::Column::CollectionId.eq(collection))
             .one(conn)
-            .await?;
+            .await?
+            .context("Solana collection not found")?;
 
-        let sc = solana_collection_model.ok_or_else(|| anyhow!("solana collection not found"))?;
+        let collection = collection_model.context("collection not found")?;
+
+        if collection.supply == Some(collection.total_mints) {
+            bail!("Collection is sold out")
+        }
```

Comment:
Since we want to do this for all blockchains can this be done in the mutation before entering the blockchain.